module/documents/advancement/trait.mjs

```diff
@@ -88,4 +199,164 @@ export default class TraitAdvancement extends Advancement {
     }
     return set;
   }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Prepare the list of available traits from which the player can choose.
+   * @param {Set<string>} [chosen]  Traits already chosen on the advancement. If not set then it will
+   *                                be retrieved from advancement's value.
+   * @returns {{choices: SelectChoices, label: string}|null}
+   */
+  async availableChoices(chosen) {
+    // TODO: Still shows "Choose 1 x" even if not possible due to mode restriction
+    let { available, choices } = await this.unfulfilledChoices(chosen);
+
+    // If all traits of this type are already assigned, then nothing new can be selected
+    if ( foundry.utils.isEmpty(choices) ) return null;
+
+    // Remove any grants that have no choices remaining
+    let unfilteredLength = available.length;
+    available = available.filter(a => a.asSet().size > 0);
+
+    // If replacements are allowed and there are grants with zero choices from their limited set,
+    // display all remaining choices as an option
+    if ( this.configuration.allowReplacements && (unfilteredLength > available.length) ) {
+      const rep = this.representedTraits();
+      return {
+        choices: choices.filter(this.representedTraits().map(t => `${t}:*`), { inplace: false }),
+        label: game.i18n.format("DND5E.AdvancementTraitChoicesRemaining", {
+          count: unfilteredLength,
+          type: Trait.traitLabel(rep.size === 1 ? rep.first() : null, unfilteredLength)
+        })
+      };
+      // TODO: This works well for types without categories like skills where it is primarily intended,
+      // but perhaps there could be some improvements elsewhere. For example, if I have an advancement
+      // that grants proficiency in the Bagpipes and allows replacements, but the character already has
+      // Bagpipe proficiency. In this example this just lets them choose from any other tool proficiency
+      // as their replacement, but it might make sense to only show other musical instruments unless
+      // they already have proficiency in all musical instruments. Might not be worth the effort.
+    }
+
+    // Create a choices object featuring a union of choices from all remaining grants
+    const remainingSet = new Set(available.flatMap(a => Array.from(a.asSet())));
+    choices.filter(remainingSet);
+
+    // Simplify label if exclusive mode and more than one set of choices still available
+    const simplifyNotification = this.configuration.choiceMode === "exclusive"
+      && (new Set(available.map(a => a._index))).size > 1;
```

Comment:
Well, the returned value with simplified notification would still need to contain the filtered `choices` value, so we could calculate `simplifyNotification` anywhere after filtering `available`, but we'd still need to continue down to the bottom.

Really all this changes is for exclusive mode showing "Choose some more Traits" rather than "Choose two more Traits" when it is impossible to calculate the number of choices that can actually be made due to exclusive mode.